source/input.cpp

```diff
@@ -738,3 +831,777 @@ extern "C" BOOL WINAPI HookGetCursorPosition(LPPOINT lpPoint)
 	static const auto trampoline = reshade::hooks::call(HookGetCursorPosition);
 	return trampoline(lpPoint);
 }
+
+extern "C" SHORT WINAPI HookGetAsyncKeyState(int vKey)
+{
+	static const auto trampoline = reshade::hooks::call(HookGetAsyncKeyState);
+	SHORT sKeyState = trampoline(vKey);
+
+	// Valid (Keyboard) Keys:  8 - 255
+	if ((vKey & 0xF8) != 0)
+	{
+		if (is_blocking_keyboard_input())
+		{
+			sKeyState = 0;
```

Comment:
This could be simplified to return zero immediately, before even calling the trampoline function, àla:
```c++
if ((vKey & 0xF8) != 0 && is_blocking_keyboard_input())
    return 0;
if (vKey < 8 && is_blocking_mouse_input())
    return 0;

static const auto trampoline = reshade::hooks::call(HookGetAsyncKeyState);
return trampoline(vKey);
```

Reply:
The motivation for calling the function first originally was that this removes a bit indicating whether the input has been seen or not. But that aspect of `Get*KeyState` is unreliable anyway.

I'll change it.